webapp/channels/src/components/global_header/left_controls/history_buttons/history_buttons.tsx

```diff
@@ -49,45 +50,28 @@ const HistoryButtons = (): JSX.Element => {
     const goBack = () => {
         trackEvent('ui', 'ui_history_back');
         history.goBack();
-        window.postMessage(
-            {
-                type: 'history-button',
-            },
-            window.location.origin,
-        );
+        requestButtons();
     };
 
     const goForward = () => {
         trackEvent('ui', 'ui_history_forward');
         history.goForward();
-        window.postMessage(
-            {
-                type: 'history-button',
-            },
-            window.location.origin,
-        );
+        requestButtons();
     };
 
-    const handleButtonMessage = useCallback((message: {origin: string; data: {type: string; message: {enableBack: boolean; enableForward: boolean}}}) => {
-        if (message.origin !== window.location.origin) {
-            return;
-        }
-
-        switch (message.data.type) {
-        case 'history-button-return': {
-            setCanGoBack(message.data.message.enableBack);
-            setCanGoForward(message.data.message.enableForward);
-            break;
-        }
-        }
-    }, []);
+    const requestButtons = () => {
+        getBrowserHistoryStatus().then(({canGoBack, canGoForward}) => updateButtons(canGoBack, canGoForward));
+    };
+
+    const updateButtons = (enableBack: boolean, enableForward: boolean) => {
+        setCanGoBack(enableBack);
+        setCanGoForward(enableForward);
+    };
 
     useEffect(() => {
-        window.addEventListener('message', handleButtonMessage);
-        return () => {
-            window.removeEventListener('message', handleButtonMessage);
-        };
-    }, [handleButtonMessage]);
+        const off = onBrowserHistoryStatusUpdated(updateButtons);
+        return off;
+    }, []);
```

Comment:
Yeah I was kind of wondering this myself. I wanted a way to turn off the listeners (as I mentioned in an above comment) but I didn't want to spend too much time designing it since this functionality won't be necessary with the legacy stuff.

I'm open to ideas on how to improve it.

Reply:
It is still a listener, so... why not listen?

Reply:
Well we're not returning the listener though, we're returning the function to turn it off, hence the name `off` :P

Reply:
The way most of this kind of things is done in mobile is the following:
- `DeviceEventEmitter.addListener` returns a `EmitterSubscription`, which has a `remove` function (among other things)

This at the end is the same... you are adding a subscription, and you are receiving the means to remove it.

So I would read it more or less fine if it was something like...
`const removeListener = addListenerOnBrowserHistoryStatusUpdated(updateButtons);`

But... yeah, not sure either. 0/5 on this for sure.

Reply:
Gotcha. So `EventEmitter` supports `on` and `off`, which is where I got this from: https://nodejs.org/api/events.html#emitteroneventname-listener

Also 0/5.

Reply:
Then let's wait for @hmhealey review so he can break the tie. Or bring a better idea :stuck_out_tongue:

Reply:
Yeah that could work. I'll see what I can do.

Reply:
So thinking on this I'm not sure just adding `useDesktopListener = useEffect` is really necessary. I suppose it makes things a little clearer, but seems like we're just adding some bloat for that.

I've added a comment for this case, but maybe this is just my bias but I think the code is relatively self-explanatory. `on<doAThing>` always returns `off<doAThing>` which stops the listener. That feels like a pretty common convention to me.

I think this is something we can iterate on going forward, but since a) this is only one case so far and b) we don't really have full context as to the use of listeners yet, I'd rather not prescribe anything.

I'm going to say 2/5 on this right now.